src/lib/y2storage/shadower.rb

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/subvol_specification"
+require "y2storage/mountable"
+
+module Y2Storage
+  # Class to manage the shadowing of Btrfs subvolumes
+  #
+  # This class is used to automatically shadow Btrfs Subvolumes. In some cases, Btrfs subvolumes are
+  # automatically added when creating a Btrfs. Those subvolumes should also be automatically hidden when
+  # other device shadows them. Moreover, subvolumes added by the user should be automatically mounted or
```

Comment:
subvolumes added by the user and also subvolumes that come from probing. Isn't it?

Reply:
Yes, right.